llvm/lib/Transforms/Vectorize/VPlanTransforms.cpp

```diff
@@ -4617,10 +4619,11 @@ void VPlanTransforms::materializeVectorTripCount(VPlan &Plan,
                                                  bool TailByMasking,
                                                  bool RequiresScalarEpilogue) {
   VPValue &VectorTC = Plan.getVectorTripCount();
-  assert(VectorTC.isLiveIn() && "vector-trip-count must be a live-in");
+  assert(isa<VPSymbolicValue>(VectorTC) &&
+         "vector-trip-count must be a live-in");
   // There's nothing to do if there are no users of the vector trip count or its
   // IR value has already been set.
-  if (VectorTC.getNumUsers() == 0 || VectorTC.getLiveInIRValue())
+  if (VectorTC.getNumUsers() == 0 || VectorTC.getUnderlyingValue())
```

Comment:
getUnderlyingValue() seems to be a replacement for getLiveInIRValue()?

Reply:
for the symbolic case yes, until we properly avoid setting the underlying value